articles/ai-services/openai/includes/spring.md

```diff
@@ -0,0 +1,246 @@
+```
+
+The generated files and folders will resemble the following structure:
+
+```
+ai-completion-demo/
+|-- pom.xml
+|-- mvn
+|-- mvn.cmd
+|-- HELP.md
+|-- src/
+    |-- main/
+    |   |-- resources/
+    |   |   |-- application.properties
+    |   |-- java/
+    |       |-- com/
+    |           |-- example/
+    |               |-- aicompletiondemo/
+    |                   |-- AiCompletionApplication.java
+    |-- test/
+        |-- java/
+            |-- com/
+                |-- example/
+                    |-- aicompletiondemo/
+                        |-- AiCompletionApplicationTests.java
+``` 
+
+## Edit Spring application
+
+
+1. Edit pom.xml file.
+
+    From the root of the project directory, open the *pom.xml* file in your preferred editor or IDE and overwrite the file with following content: 
+
+    ```xml
+    <?xml version="1.0" encoding="UTF-8"?>
+    <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    	<modelVersion>4.0.0</modelVersion>
+    	<parent>
+    		<groupId>org.springframework.boot</groupId>
+    		<artifactId>spring-boot-starter-parent</artifactId>
+    		<version>3.2.0</version>
+    		<relativePath/> <!-- lookup parent from repository -->
+    	</parent>
+    	<groupId>com.example</groupId>
+    	<artifactId>ai-completion-demo</artifactId>
+    	<version>0.0.1-SNAPSHOT</version>
+    	<name>AICompletion</name>
+    	<description>Demo project for Spring Boot</description>
+    	<properties>
+    		<java.version>17</java.version>
+    	</properties>
+    	<dependencies>
+    		<dependency>
+    			<groupId>org.springframework.boot</groupId>
+    			<artifactId>spring-boot-starter</artifactId>
+    		</dependency>
+    		<dependency>                 
+    		    <groupId>org.springframework.experimental.ai</groupId>
+    		    <artifactId>spring-ai-azure-openai-spring-boot-starter</artifactId>
+    		    <version>0.7.0-SNAPSHOT</version>
+    		</dependency>
+    		<dependency>
+    			<groupId>org.springframework.boot</groupId>
+    			<artifactId>spring-boot-starter-test</artifactId>
+    			<scope>test</scope>
+    		</dependency>
+    	</dependencies>
+    	<build>
+    		<plugins>
+    			<plugin>
+    				<groupId>org.springframework.boot</groupId>
+    				<artifactId>spring-boot-maven-plugin</artifactId>
+    			</plugin>
+    		</plugins>
+    	</build>
+    	<repositories>
+    		<repository>
+    			<id>spring-snapshots</id>
+    			<name>Spring Snapshots</name>
+    			<url>https://repo.spring.io/snapshot</url>
+    			<releases>
+    				<enabled>false</enabled>
+    			</releases>
+    		</repository>
+    	</repositories>
+    </project>
+    ```
+
+1. From the *src/main/java/com/example/aicompletiondemo* folder, open *AiCompletionApplication.java* in your preferred editor or IDE and paste in the following code.
+
+    
+    ```java
```

Comment:
Comment from @mrbullwinkle:

Same comment as on the chat code example, I don't see where you are setting the model deployment name/telling the customer where to set this info.

Same comment on environment variables but I am assuming those are set automatically under the hood by the AiClient if the environment variables of the same name are created.

Reply:
The `[!INCLUDE [environment-variables](spring-environment-variables.md)]` includes documentation and instruction on how you set the environment variables for the endpoints.  The same type of doc should be in each sample.

Spring AI auto configuration reads these env variables and sets them for you, so no need to do it in code.